corehq/apps/dump_reload/couch/load.py

```diff
@@ -47,13 +47,13 @@ def load_objects(self, object_strings, force=False, dry_run=False):
     def _doc_type_matches_filter(self, doc_type):
         return not self.object_filter or self.object_filter.findall(doc_type)
 
-    def _get_db_for_doc_type(self, doc_type):
+    def _get_db_for_doc_type(self, doc_type, chunksize=None):
         if doc_type not in self._dbs:
             couch_db = get_db_by_doc_type(doc_type)
             if couch_db is None:
                 raise DocumentClassNotFound('No Document class with name "{}" could be found.'.format(doc_type))
             callback = LoaderCallback(self._success_counter, self.stdout)
-            db = IterDB(couch_db, new_edits=False, callback=callback)
+            db = IterDB(couch_db, new_edits=False, callback=callback, chunksize=chunksize)
```

Comment:
Guessing this is used specifically for applications - they're MUCH much bigger than anything else.  Another approach to consider is to pass `chunksize=1` if it's an app (as determined by `doc_type`).  Then this doesn't need to be controlled by the user, and everything else can still be loaded in chunks for speed.

Reply:
I agree with Ethan, although I'm conflicted.

It is likely although not guaranteed that one or a very few applications are the culprits when the default chunk size is a problem. It would be nice to automatically fall back to a small chunk size automatically and temporarily on an as-needed basis within the loader loop. Is it possible to detect when the chunk size is too large by catching a specific error from Couch or something like that? Or does a too-large chunk cause a too-generic-to-catch error like `MemoryError` or something along those lines?

Reply:
From the screenshots the customer sent, the error they get from couch is:
`requests.exceptions.HTTPError: Problem loading data <zip file>: 413 Client Error: Request Entity Too Large for url`
which I guess it's something we could catch for. So once caught, should it try saving increasingly smaller portions of the chunk its trying to save/delete (halving the list until it reaches 1 item for example)? Is there a way for myself or QA to test something like this?

Reply:
Not sure how best to test it, but we've certainly seen this before for apps.  It's not super elegant, but IMO saving all apps with a chunksize of 1 would be sufficient to solve the issue and not overly hinder the performance.

It'd be really cool if `IterDB` could be retooled to keep track of the chunk size in bytes and commit automatically based on size rather than doc count.  That sounds like maybe more trouble than it's worth, though.